nemo/core/classes/modelPT.py

```diff
@@ -294,7 +294,11 @@ def restore_from(
         if save_restore_connector is None:
             save_restore_connector = SaveRestoreConnector()
 
-        restore_path = os.path.abspath(os.path.expanduser(restore_path))
+        if save_restore_connector.model_extracted_dir is None:
+            restore_path = os.path.abspath(os.path.expanduser(restore_path))
+        else:
+            restore_path = os.path.abspath(os.path.expanduser(save_restore_connector.model_extracted_dir))
```

Comment:
what if the user provided absolute path outside of their homedir?

Reply:
expanduser() only expands `~` in a user provided path, so if absolute path is given then as long as theres no ~ in it, it will be correct expansion.